semeio/workflows/misfit_preprocessor/cluster_config.py

```diff
@@ -0,0 +1,207 @@
+            lower,
+        ),
+        ">"
+        if upper is None
+        else "{}{}".format(
+            upper,
+            "]" if upper_inclusive else ")",
+        ),
+    )
+
+    @cs.validator_msg(msg)
+    def validator(value):
+        lower_valid = lower is None or (
+            lower <= value if lower_inclusive else lower < value
+        )
+        upper_valid = upper is None or (
+            value <= upper if upper_inclusive else value < upper
+        )
+        return lower_valid and upper_valid
+
+    return validator
+
+
+@cs.validator_msg("t must be an integer if a max cluster criteria is choosen")
+def _t_is_int_if_maxclust(spearman_config):
+    if spearman_config[_CRITERION] in (_MAXCLUST, _MAXCLUST_MONOCRIT):
+        return isinstance(spearman_config[_SPEARMAN_THRESHOLD], int)
+    return True
+
+
+@cs.transformation_msg(
+    (
+        "Will be defaulted to 5 for all max cluster criterias and "
+        "1.15 for all other criterias"
+    )
+)
+def _inject_default_t(config):
+    config = copy.deepcopy(config)
+    criterion = config.get(_CRITERION)
+    if _SPEARMAN_THRESHOLD in config:
+        pass
+    elif criterion in (_MAXCLUST, _MAXCLUST_MONOCRIT):
+        config[_SPEARMAN_THRESHOLD] = 5
+    elif criterion in (_INCONSISTENT, _DISTANCE, _MONOCRIT):
+        config[_SPEARMAN_THRESHOLD] = 1.15
+    else:
+        raise ValueError("Unknown criterion: {}".format(criterion))
+    return config
+
+
+HIERARCHICAL_SCHEMA = {
+    cs.MetaKeys.Type: cs.types.NamedDict,
+    cs.MetaKeys.Content: {
+        "hierarchical": {
+            cs.MetaKeys.ElementValidators: (_t_is_int_if_maxclust,),
+            cs.MetaKeys.Transformation: _inject_default_t,
+            cs.MetaKeys.Type: cs.types.NamedDict,
+            cs.MetaKeys.Description: (
+                "The {linkage} implementation is backed by scipy and for a "
+                "more detailed description we refer the reader to the "
+                "documentation of scipy.cluster.hierarchy.linkage` "
+                "(https://docs.scipy.org/doc/scipy/reference/generated"
+                "/scipy.cluster.hierarchy.linkage.html)."
+                "The {fcluster} implementation is backed by scipy and for a "
+                "more detailed description we refer the reader to the "
+                "documentation of scipy.cluster.hierarchy.fcluster` "
+                "(https://docs.scipy.org/doc/scipy/reference/generated"
+                "/scipy.cluster.hierarchy.fcluster.html)."
+            ).format(linkage=_LINKAGE, fcluster=_FCLUSTER),
+            cs.MetaKeys.Content: {
+                _SPEARMAN_THRESHOLD: {
+                    cs.MetaKeys.Type: cs.types.Number,
+                    cs.MetaKeys.Description: (
+                        "Scalar threshold for the clustering. When a "
+                        "'{maxclust}' {criterion} is used, the scalar gives "
+                        "the maximum number of clusters to be formed."
```

Comment:
There should be a space after this sentence (and the three other places this occurs)